### ISSUE TITLE
Reduce size of distributed Linux binaries

### ISSUE THREAD

**Issue by janisozaur** [OpenRCT2/OpenRCT2#6232]
**OS:** Linux

The compiled binary for Linux x86-64 of OpenRCT2 takes a lot of space, ~40MiB unpacked for just the ELF binary. This is partially because we use `RelWithDebInfo` build type, which embeds debug symbols into the file itself. This is very useful when we need to do actual debugging, but in most cases it just takes space.

The DWARF debug information, among others, has two types of informations:
* code-to-source-line mappings
* variable tracking information

For user to be able to provide valuable stack traces (symbolicated, with line numbers), only the first part is needed\* and the variable tracking can be turned off with `-fno-var-tracking` GCC option and this results in 25% smaller binary (30MiB). Unfortunately, disabling it at compilation time makes debug symbols much less useful.

Instead, I would like to try removing it in a post-process step. It looks like variable tracking information is located in `.debug_loc` section and perhaps simply extracting it out from the binary and into separate symbol file would yield desired effect:
* having smaller distributed binaries
* not sacrificing value of debug information.

\* This is just my assumption for now, I still have to verify it.

**Comment by IntelOrca** [comment]
Can you do what we do for Windows and distribute the debug information as a separate file?

**Comment by janisozaur** [comment]
> Can you do what we do for Windows and distribute the debug information as a separate file?

Yes, that's what I want to achieve, only with some debug information still in the binary. That way if a user experiences a crash, he can still provide exact stack trace with source lines.

Function names can be left in (and in cases when they are not `static` they have to be left in), even when other debug information is not present.

**Comment by kiroma** [comment]
Would that make any difference?

**Comment by baryluk** [comment]
There is an additional issue I just spotted.

I was looking at `OpenRCT2-0.2.3-develop-43cb570ae-linux-i686.tar.gz` and
`OpenRCT2-0.2.3-develop-43cb570ae-linux-x86_64.tar.gz`

The archive sizes are already raising eyebrows:

Linux amd64 - ~31.6MB ; Linux i686 ~96.5MB

Looking at binaries after unpacking:

Linux amd64 are ~33.4MB ; Linux i686 builds are ~63MB

Additionally in .gz file, I do see:

* in Linux i686, `lib/libopenrct2.a`  (for static linking, with all compiled object files; they are all with debug_info, not stripped), with size of 119MB
* in Linux amd64, `libopenrct2.so` (with debug_info, not stripped) with size of 30MB

I am pretty sure `lib/libopenrct2.a` is just a mistake to be included in the archive?

The difference between statically vs dynamically linking of libopenrct2 between two archs is also worrying, and is probably a mistake in some script.

**Comment by janisozaur** [comment]
The difference stems from the constraints we set about having a small enough amount of jobs on Travis-CI, each of which takes quite some time to build, and wanting to cover all possible configurations. Thanks for bringing this up, perhaps we should change the build configurations a bit. Feel free to come up with some proposals. And yes, the static archive is not something that should end up in our redists.
